addons/account/models/account_move.py

```diff
@@ -1338,56 +1338,128 @@ class AccountPartialReconcile(models.Model):
         help='Utility field to express amount currency')
     company_id = fields.Many2one('res.company', related='debit_move_id.company_id', store=True, string='Currency')
 
-    def create_exchange_rate_entry(self, aml_to_fix, amount_diff, diff_in_currency, currency, move_date):
-        """ Automatically create a journal entry to book the exchange rate difference.
-            That new journal entry is made in the company `currency_exchange_journal_id` and one of its journal
-            items is matched with the other lines to balance the full reconciliation.
+    @api.multi
+    def _prepare_exchange_diff_line_to_reconcile(self, amount_diff, currency,
+                                            diff_in_currency, move):
+        self.ensure_one()
+        res = {
+            'name': _('Currency exchange rate difference'),
+            'debit': amount_diff < 0 and -amount_diff or 0.0,
+            'credit': amount_diff > 0 and amount_diff or 0.0,
+            'account_id': self.debit_move_id.account_id.id,
+            'move_id': move.id,
+            'currency_id': currency.id,
+            'amount_currency': -diff_in_currency,
+            'partner_id': self.debit_move_id.partner_id.id,
+        }
+        return res
+
+    @api.multi
+    def _prepare_exchange_diff_move_line(self, amount_diff, currency,
+                                         diff_in_currency, move):
+        self.ensure_one()
+        exchange_journal = self.company_id.currency_exchange_journal_id
+        res = {
+            'name': _('Currency exchange rate difference'),
+            'debit': amount_diff > 0 and amount_diff or 0.0,
+            'credit': amount_diff < 0 and -amount_diff or 0.0,
+            'account_id':  amount_diff > 0 and
+            exchange_journal.default_debit_account_id.id or
```

Comment:
if you absolutely want to keep on multiple lines, at least increase the indentation